src/prefect/futures.py

```diff
@@ -431,9 +444,11 @@ def resolve_futures_to_states(
 
     Unsupported object types will be returned without modification.
     """
-    futures: Set[PrefectFuture[R]] = set()
+    futures: set[PrefectFuture[R]] = set()
 
-    def _collect_futures(futures, expr, context):
+    def _collect_futures(
+        futures: set[PrefectFuture[R]], expr: Any, context: Any
```

Comment:
`context` must at least support `.get(str)` so I think it should be a `Mapping[str, Any]`, certainly given what I've seen of how contexts are used in perfect. Ditto in other argument annotations below.